### PR TITLE
Config and refinements

### PR DESCRIPTION
**Pull request by dkosowski87**
Updated package.json
Added editorconfig
Added overcommit with eslint
Added npmignore
Added travis config
Refined test (eslint)
Added contributing guidelines.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+cache:
+  directories:
+    - node_modules
+
+install:
+  - npm install
+
+script:
+  - npm test
```
Comment (RafalWilinski) [inline .travis.yml]: Why travis? I thought we're using circleCI

Comment (dkosowski87) [inline .travis.yml]: @RafalWilinski To be honest I was pretty sure that all our open source libraries where using travis. Maybe I got the wrong impression 🤔

File: test/mapFilter.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ describe('#mapFilter()', function () {
     shouldNeverSpam: 'true',
   };
 
-  it('maps a provided filter entity to a filter attributes object', function () {
+  it('maps a provided filter entity to a filter attributes object', () => {
```
Comment (netguru-hound) [inline test/mapFilter.js]: 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: test/mapFilter.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
 const expect = require('chai').expect;
 const mapFilter = require('../lib/mapFilter');
 
-describe('#mapFilter()', function () {
+describe('#mapFilter()', () => {
```
Comment (netguru-hound) [inline test/mapFilter.js]: 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

**Pull request merged**